File: subprojects/frida-python/examples/inject_library/inject_blob.py
#
# Compile example.dylib like this:
# $ clang -shared example.c -o example.dylib
#
# Then run:
# $ python inject_blob.py Twitter example.dylib
#

import sys

import frida


def on_uninjected(id):
    print("on_uninjected id=%u" % id)


target, library_path = sys.argv[1:]

device = frida.get_local_device()
device.on("uninjected", on_uninjected)
with open(library_path, "rb") as library_file:
    library_blob = library_file.read()
id = device.inject_library_blob(target, library_blob, "example_main", "w00t")
print("*** Injected, id=%u -- hit Ctrl+D to exit!" % id)
sys.stdin.read()

File: subprojects/frida-node/examples/inject_library/inject_blob.ts
/*
 * Compile example.dylib like this:
 * $ clang -shared example.c -o example.dylib
 *
 * Then run:
 * $ node inject_blob.js Twitter example.dylib
 */

import frida from "frida";
import fs from "fs/promises";

const [target, libraryPath] = process.argv.slice(2);

let device: frida.Device | null = null;

async function main() {
    const libraryBlob = await fs.readFile(libraryPath);

    device = await frida.getLocalDevice();
    device.uninjected.connect(onUninjected);

    try {
        const id = await device.injectLibraryBlob(target, libraryBlob, "example_main", "w00t");
        console.log("[*] Injected id:", id);
    } catch (e) {
        device.uninjected.disconnect(onUninjected);
        throw e;
    }
}

function onUninjected(id: frida.InjecteeID) {
    console.log("[*] onUninjected() id:", id);
    device!.uninjected.disconnect(onUninjected);
}

main()
    .catch(e => {
        console.error(e);
    });
